docs/javascript/v2/guides/javascript-quickstart.mdx

```diff
@@ -310,89 +988,76 @@ muteVid.addEventListener("click", () => {
+    text-transform: uppercase;
+    letter-spacing: 1px;
+    border: 2px solid #37474f;
+    width: 64px;
+    height: 64px;
+    border-radius: 50%;
+    text-align: center;
+    background-color: #607d8b;
+    box-shadow: 0 0 10px rgba(0, 0, 0, 0.4);
+    color: white;
+  }
 ```
 
-![Mute/Unmute](/guides/mute-unmute.gif)
-
-## Mirroring local video
+</Tab>
 
-Each `peer` object has an `isLocal` property that indicates whether the peer is local or remote.
-When looping over the peer list we can use this property to decide whether to mirror the video or not.
+### Step 6: Change UI based on connection state
 
-<Tabs id="mirror" items={['JS', 'CSS']} />
+Right now, our join form will show up even after we have joined the room. We need a way to know the connection state of the room and hide the form after we've connected.
 
-<Tab id="mirror-0" >
+We can do this by subscribing to the store with the [selectIsConnectedToRoom](/api-reference/javascript/v2/home/content#select-is-connected-to-room) selector from `hmsStore`.
 
-```jsx{10,28}:script.js section=MirroringLocalVideo sectionIndex=1 tab=JS
-function renderPeers(peers) {
-  peersContainer.innerHTML = "";
-
-  if (!peers) {
-    peers = hmsStore.getState(selectPeers);
+```jsx:index.js section=ChangingUIBasedOnConnectionState sectionIndex=1
+// Showing the required elements on connection/disconnection
+function onConnection(isConnected) {
+  if (isConnected) {
+    form.classList.add("hide");
+    conference.classList.remove("hide");
+    leaveBtn.classList.remove("hide");
+    controls.classList.remove("hide");
+  } else {
+    form.classList.remove("hide");
+    conference.classList.add("hide");
+    leaveBtn.classList.add("hide");
+    controls.classList.add("hide");
   }
-
-  peers.forEach((peer) => {
-    const video = h("video", {
-      class: "peer-video" + (peer.isLocal ? " local" : ""),
-      autoPlay: true,
-      muted: true
-    });
-
-    hmsActions.attachVideo(peer.videoTrack, video);
-
-    const peerContainer = h(
-      "div",
-      {
-        class: "peer-container"
-      },
-      video,
-      h(
-        "div",
-        {
-          class: "peer-name"
-        },
-        peer.name + (peer.isLocal ? " (You)" : "")
-      )
-    );
-
-    peersContainer.append(peerContainer);
-  });
 }
-```
 
-</Tab>
-
-<Tab id="mirror-1">
-
-```css:styles.css section=MirroringLocalVideo sectionIndex=1 tab=CSS
-.local.peer-video {
-    transform: scaleX(-1);
-}
+// Listen to the connection state
+hmsStore.subscribe(onConnection, selectIsConnectedToRoom);
 ```
 
-</Tab>
+Refer to [Test the app section](/javascript/v2/guides/javascript-quickstart#test-the-app) above to test your sample app locally.
 
-That wraps it up for this guide. We hope you had fun.
-There are plenty of cool stuff which can be done with our SDK, be sure to check the [features section](/javascript/v2/features/integration) for more information.
+## Next steps
 
 Here's the complete example.
 
 <Codesandbox id="5rmes" />
```

Comment:
thoughts on embedding this on the top where we mention codesandbox as well?

Reply:
done